plone/app/uuid/utils.py

```diff
@@ -1,18 +1,33 @@
 # -*- coding: utf-8 -*-
 from Products.CMFCore.utils import getToolByName
+from Products.ZCatalog.query import IndexQuery
 from zope.component.hooks import getSite
 
 
 def uuidToPhysicalPath(uuid):
     """Given a UUID, attempt to return the absolute path of the underlying
     object. Will return None if the UUID can't be found.
-    """
 
-    brain = uuidToCatalogBrain(uuid)
-    if brain is None:
+    This version is four times faster than the original.
+    """
+    site = getSite()
+    if site is None:
         return None
-
-    return brain.getPath()
+    catalog = getToolByName(site, 'portal_catalog', None)
+    if catalog is None:
+        return None
+    index = catalog.Indexes["UID"]
```

Comment:
Do we gain speed by caching the index i.e. on request? (which should not cause any harm)

Reply:
I looked at caching the index on the request, but when you have a match for a uuid to an `rid`, then you still need to get the catalog to get the path for this `rid`.
So I now cache the portal_catalog on `request._catalog`. Seems fine, and indeed another speedup.

Reply:
Caching the catalog on request._catalog is probably a good idea overall in Plone (not only here). But that's a different topic.